src/hacking/building-for-openharmony.md

```diff
@@ -105,15 +110,84 @@ Depending on your Linux distributions you may want to use a different group.
 To check if `hdc` is now working, you can run `hdc list targets` and it should show your device serial number.
 If it doesn't work, try rebooting.
 
-Please note, that your phone needs to be in "Developer mode" with USB debugging enabled.
+Please note, that your device needs to be in "Developer mode" with USB debugging enabled.
 The process here is exactly the same as one android:
 1. Tap the build number multiple times to enable developer mode.
 2. Then navigate to the developer options and enable USB debugging.
 3. When you connect your device for the first time, confirm the pop-up asking you if you want to trust the computer you are connecting to.
 
+## Signing configuration
+
+Most devices require that the HAP is digitally signed by the developer to be able to install it.
+When using the `hvigor` tool, this can be accomplished by setting a static `signingConfigs` object in the `build-profile.json5` file or
+by dynamically creating the `signingConfigs` array on the appication context object using the `hvigorfile.ts` script.
+
+The `signingConfigs` property is an array of objects with the following structure:
+
+```json
+
+{
+    "name": "default",
+    "type": "OpenHarmony",
+    "material": {
+        "certpath": "/path/to/app-signing-certificate.cer",
+        "storePassword": "<encrypted password>",
+        "keyAlias": "debugKey",
+        "keyPassword": "<encrypted password>",
+        "profile": "/path/to/signed-profile-certificate.p7b",
+        "signAlg": "SHA256withECDSA",
+        "storeFile": "/path/to/java-keystore-file.p12"
+    }
+}
+```
+
+Here <encrypted password> is a hexadecimal string representation of the plaintext password after being encrypted.
+The key and salt information used to encrypt these passwords are generated by DevEco Studio IDE and are stored on-disk alongside
+the certificate files and keystore, usually under `<USER HOME>/.ohos/config/openharmony`.
+
+You can use the IDE to generate information needed for password encryption, the required application and
+profile certificate files, and the keystore itself using:
+
+1. Open Project Structure dialog from `File > Project Structure` menu.
+2. Under the 'Signing Config' tab, enable the 'Automatically generate signature'.
+
+Once generated, it is necessary to point `mach` to the above "signing material" configuration using the
+`SERVO_OHOS_SIGNING_CONFIG` environment variable. The value of the variable must be a file path to a
+valid `.json` file with the same structure as the `signingConfigs` property given above, but with `certPath`, `storeFile` and
+`profile` given as *paths relative to the json file*, instead of absolute paths.
+
+
+## Building Servo shell
+
+Before building servo, ensure the following environment variables are set in the current shell:
+
+```commandline
+    # Note: The openharmony sdk is under ${DEVECO_SDK_HOME}/HarmonyOS-NEXT-${HOS_VERSION}/openharmony
```

Comment:
I think in Developer Beta 7 / Beta 1 this path was changed. I'd need to check next week, but I think it's now something like `${DEVECO_SDK_HOME}/default/openharmony`.

Reply:
I've update the text to be `${DEVECO_SDK_HOME}/default/openharmony`. Is that different from one where commandline tools is used?

Reply:
`DEVECO_SDK_HOME` points to `/path/to/DevEcoStudio/sdk`, so that seems to match.